server/service/oauth.go

```diff
@@ -107,24 +107,45 @@ func (o *OAuth) GetUser(ctx context.Context, code string) (*githubUser, error) {
 		return nil, fmt.Errorf("can't parse github user response: %s", err)
 	}
 
-	user.Role = model.Requester
+	if err = o.setRole(o, client, &user); err != nil {
```

Comment:
`o.setRole(o)` looks very weird for me.
Do you think you could refactor it like this:

```go
type OAuth struct {
    // ...
	restrictionChecker restrictionChecker
}

func NewOAuth(clientID, clientSecret, restrictAccess, restrictRequesterAccess string) *OAuth {
	// ...
	return &OAuth{
        // ...
		restrictionChecker:      &githubPermissions{}
	}
}

func (o *OAuth) setRole(client *http.Client, user *githubUser) error {
	//...
	o.restrictionChecker(client, o.restrictAccess, user.Login)
	// ...
}

type githubPermissions struct {}

func (c *githubPermissions) checkAccess(client *http.Client, restriction, login string) error {
	// ...
}
```

or maybe you see how to do it even better?